app/src/test/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModelTest.kt

```diff
@@ -65,20 +71,62 @@ class ProfileViewModelTest : ShouldSpec(
 			coVerify { authRepository.updateUserAvatar(uri) }
 		}
 
-		should("load profile WHEN updateAvatar called") {
-			coEvery { authRepository.updateUserAvatar(uri) } returns profile
-
-			tested.updateAvatar(uri)
+		should("load profile WHEN specific method called") {
+			listOf(
+				ManageProfileParam(
+					{ authRepository.updateUserAvatar(uri) },
+					{ tested.updateAvatar(uri) },
+					profile,
+				),
+				ManageProfileParam(
+					{ authRepository.updateUserName("Marian K") },
+					{ tested.updateName("Marian K") },
+					profile,
+				),
+			).forAll { param ->
+				coEvery { param.profileMethod.invoke() } returns param.data as Profile
+
+				param.action.invoke()
+
+				tested.profile.value shouldBe Loaded(param.data)
+			}
+		}
 
-			tested.profile.value shouldBe Loaded(profile)
+		should("be loading WHEN specific method called and start executing") {
+			listOf(
+				ManageProfileParam(
+					{ authRepository.updateUserAvatar(uri) },
+					{ tested.updateAvatar(uri) },
+					Loading,
+				),
```

Comment:
do we need loading here as we know that we will check loading in every test ? this could consist of two first values only

Reply:
Yes, it is redundant I deleted it